### PR TITLE
Keshav | Week 5 - Counter using React-Redux

### PR DESCRIPTION
**Pull request by choudhary-keshav**
Pull Request Questionnaire

(1) Which Jira ticket(s) is this pull request for, and is the ticket now complete?

(2) Briefly describe your change.
I have integrated my week-5 assignment in this joint project. In my week-5 assignment I have added a counter component in week-5. I have developed this counter using react-redux.

(3) What parts of the application are affected by your change?
It'd just affect the "Keshav's", "week-5" component.

(4) How did you test your change? If relevant, which role(s) and/or customer configuration(s) did you use?
Tested my application on various edge cases and some common scenarios

Root Cause (if any)?
-

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -23,11 +24,12 @@ const App: React.FC = () => {
       <Routes>
         <Route path='/' element={<Home />}>
           <Route path='keshav' element={<Keshav />}>
-            <Route path='week1&2' element={<Week1 />}></Route>
-            <Route path='week3' element={<Week3Main />}></Route>
-            <Route path='week4' element={<Week4 />}>
-              <Route path=':pageId' element={<Week4 />}></Route>
+            <Route path='week1&2' element={<KeshavWeek1 />}></Route>
+            <Route path='week3' element={<KeshavWeek3 />}></Route>
```
Comment (aditya-s-19) [inline src/App.tsx]: we can use self-closing tag in these routes

File: src/components/Home/Keshav/Week5/Counter/index.tsx [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  const decrement = () => {
+    dispatch(decrementCounter());
+  };
+
+  return (
+    <CounterWrapper>
+      <h1>Counter using Redux</h1>
+      <span className='counterHeading'>Counter : {count}</span>
+      <br />
+      <br />
```
Comment (aditya-s-19) [inline src/components/Home/Keshav/Week5/Counter/index.tsx:24]: use css to add spacing instead of br tags

File: src/components/Home/Keshav/Week5/styles.ts [hunk 3]
```diff
@@ -0,0 +1,3 @@
+import styled from 'styled-components';
+
+export const Week5Wrapper = styled.div``;
```
Comment (aditya-s-19) [inline src/components/Home/Keshav/Week5/styles.ts]: this file seems unnecessary if no styling is needed

File: src/components/Home/Keshav/Week5/Counter/styles.ts [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import styled from 'styled-components';
+
+export const CounterWrapper = styled.div`
+  .counterHeading {
+    padding: 10px;
+    background-color: #000000;
+    color: #fff;
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Keshav/Week5/Counter/styles.ts]: diff color code format

File: src/components/Home/Keshav/Week5/Counter/styles.ts [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    &:hover {
+      background-color: #fff;
+      color: #000000;
+      transition-duration: 1s;
+    }
+    &:active {
+      transform: scale(0.9);
+    }
+  }
+`;
```
Comment (RitikaGarg-ITT) [inline src/components/Home/Keshav/Week5/Counter/styles.ts:23]: check the color code format within the complete file

Comment (choudhary-keshav) [reply src/components/Home/Keshav/Week5/Counter/styles.ts:23]: Done!

File: src/components/Home/Keshav/Week5/Counter/styles.ts (hunk 4 above)
Comment (choudhary-keshav) [reply src/components/Home/Keshav/Week5/Counter/styles.ts]: Done!

File: src/components/Home/Keshav/Week5/Counter/index.tsx (hunk 2 above)
Comment (choudhary-keshav) [reply src/components/Home/Keshav/Week5/Counter/index.tsx:24]: I have tried giving margin-bottom directly to span tag, but content was not coming below to it.

File: src/App.tsx (hunk 1 above)
Comment (choudhary-keshav) [reply src/App.tsx]: Done!

File: src/components/Home/Keshav/Week5/Counter/index.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  const dispatch = useDispatch();
+
+  const increment = () => {
+    dispatch(incrementCounter());
+  };
+
+  const decrement = () => {
+    dispatch(decrementCounter());
+  };
+
```
Comment (shreyansh-khatri) [inline src/components/Home/Keshav/Week5/Counter/index.tsx:18]: Please specify type for all functions

File: src/components/Home/Keshav/Week5/redux/store/store.ts [hunk 7]
```diff
@@ -0,0 +1,6 @@
+import { createStore } from 'redux';
+import CounterReducer from '../reducers/CounterReducer';
+
+const store = createStore(CounterReducer);
+
```
Comment (shreyansh-khatri) [inline src/components/Home/Keshav/Week5/redux/store/store.ts:5]: create store is deprecated now, try using configure store

File: src/components/Home/Keshav/Week5/Counter/index.tsx [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+import { RootState } from '../redux/actions/types';
+
+const Counter: React.FC = () => {
+  const count = useSelector((state: RootState) => state.count);
+  const dispatch = useDispatch();
+
+  const increment = () => {
+    dispatch(incrementCounter());
+  };
+
```
Comment (shreyansh-khatri) [inline src/components/Home/Keshav/Week5/Counter/index.tsx:14]: since it is single line, you can use inline function here.

**Review by shreyansh-khatri — CHANGES_REQUESTED** [review]

File: src/components/Home/Keshav/Week5/redux/store/store.ts (hunk 7 above)
Comment (choudhary-keshav) [reply src/components/Home/Keshav/Week5/redux/store/store.ts:5]: I have used 'redux' for my week-5 assignment. "configureStore()" function is provided by "redux toolkit".

File: src/components/Home/Keshav/Week5/Counter/index.tsx (hunk 8 above)
Comment (choudhary-keshav) [reply src/components/Home/Keshav/Week5/Counter/index.tsx:14]: It is the standard way, I have seen across various tutorials and websites and I think it's better to keep separate function for dispatching actions, so that same can be reused if we want it multiple times in a file.

File: src/components/Home/Keshav/Week5/Counter/index.tsx (hunk 6 above)
Comment (choudhary-keshav) [reply src/components/Home/Keshav/Week5/Counter/index.tsx:18]: Done!

**Review by RitikaGarg-ITT — APPROVED** [review]

**Review by aditya-s-19 — APPROVED** [review]

**Review by shreyansh-khatri — APPROVED** [review]

File: src/components/Home/Keshav/Week5/Counter/index.tsx [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+  const decrement = (): void => {
+    dispatch(decrementCounter());
+  };
+
+  return (
+    <CounterWrapper>
+      <h1>Counter using Redux</h1>
+      <span className='counterHeading'>Counter : {count}</span>
+      <br />
```
Comment (anusreesubash) [inline src/components/Home/Keshav/Week5/Counter/index.tsx:23]: please use css if you want to add extra lines in between elements. you can use flex to display inline elements in blocks.
break tag is commonly used where you want to create a line break within a paragraph or text.

**Review by shreegovinda — APPROVED** [review]
